src/internal/operators/throttleTime.ts

```diff
@@ -90,105 +89,113 @@ export function throttleTime<T>(
   scheduler: SchedulerLike = asyncScheduler,
   config: ThrottleConfig = defaultThrottleConfig
 ): MonoTypeOperatorFunction<T> {
-  return (source: Observable<T>) => lift(source, new ThrottleTimeOperator(duration, scheduler, !!config.leading, !!config.trailing));
-}
-
-class ThrottleTimeOperator<T> implements Operator<T, T> {
-  constructor(private duration: number, private scheduler: SchedulerLike, private leading: boolean, private trailing: boolean) {}
-
-  call(subscriber: Subscriber<T>, source: any): TeardownLogic {
-    return source.subscribe(new ThrottleTimeSubscriber(subscriber, this.duration, this.scheduler, this.leading, this.trailing));
-  }
-}
-
-class ThrottleTimeSubscriber<T> extends Subscriber<T> {
-  private throttled: Subscription | null = null;
-  private trailingValue: T | null = null;
-  private hasTrailingValue = false;
-  private isComplete = false;
-
-  constructor(
-    destination: Subscriber<T>,
-    private duration: number,
-    private scheduler: SchedulerLike,
-    private leading: boolean,
-    private trailing: boolean
-  ) {
-    super(destination);
-  }
-
-  protected _next(value: T) {
-    const { destination } = this;
-    if (this.throttled) {
-      // There is already a throttle timer going,
-      // we don't emit in this case.
-      if (this.trailing) {
-        // Update the trailing value, so that when the
-        // throttle executes, it has a trailing value to emit.
-        this.trailingValue = value;
-        this.hasTrailingValue = true;
-      }
-    } else {
-      if (this.leading) {
-        // With the leading behavior, if we're not
-        // throttled we can emit the value right away,
-        // but then we have to start the throttle.
-        destination.next(value);
-      } else if (this.trailing) {
-        // We're NOT leading, but we are trailing, then
-        // we need to set the trailing value so once the throttle
-        // is done we can emit it. We don't do this if we're leading
-        // as well, because that would result in a double emission.
-        this.trailingValue = value;
-        this.hasTrailingValue = true;
-      }
+  return (source: Observable<T>) =>
+    lift(source, function (this: Subscriber<T>, source: Observable<T>) {
+      const subscriber = this;
+      const { leading = false, trailing = false } = config;
```

Comment:
nitpick: `const { leading = false, trailing = false } = config;` could be moved to the up to the enclosing scope - i.e. above the `return (source: ...) =>` - as the config only needs to be read once.

Reply:
Good catch